service/src/main/java/uk/gov/hmcts/reform/fpl/validation/validators/IsValidHearingEditValidator.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.reform.fpl.validation.validators;
+
+import uk.gov.hmcts.reform.fpl.enums.HearingOptions;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.validation.interfaces.IsValidHearingEdit;
+
+import java.util.List;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+import static uk.gov.hmcts.reform.fpl.enums.HearingOptions.ADJOURN_HEARING;
+import static uk.gov.hmcts.reform.fpl.enums.HearingOptions.EDIT_HEARING;
+import static uk.gov.hmcts.reform.fpl.enums.HearingOptions.VACATE_HEARING;
+
+public class IsValidHearingEditValidator implements ConstraintValidator<IsValidHearingEdit, CaseData> {
+    @Override
+    public boolean isValid(CaseData caseData, ConstraintValidatorContext constraintValidatorContext) {
```

Comment:
maybe shoreter version:
@Override
    public boolean isValid(CaseData caseData, ConstraintValidatorContext constraintValidatorContext) {
        return !(isInValidHearingEdit(caseData) || isInvalidHearingAdjournment(caseData) || isInvalidHearingVacate(caseData));
    }

    private boolean isInValidHearingEdit(CaseData caseData) {
        return EDIT_HEARING.equals(caseData.getHearingOption()) && isEmpty(caseData.getFutureHearings());
    }

    private boolean isInvalidHearingAdjournment(CaseData caseData) {
        return ADJOURN_HEARING.equals(caseData.getHearingOption()) && isEmpty(caseData.getPastAndTodayHearings());
    }

    private boolean isInvalidHearingVacate(CaseData caseData) {
        return VACATE_HEARING.equals(caseData.getHearingOption()) && isEmpty(caseData.getFutureAndTodayHearings());
    }


if you pass CaseData as single param to private method then it is not possible to use method incorrectly , like passing ADJOURN_HEARING and getFutureHearings()